tests/network/hotplug.go

```diff
@@ -434,20 +434,21 @@ var _ = SIGDescribe("nic-hotunplug", func() {
 			removeIfaceOpts := &v1.RemoveInterfaceOptions{
 				Name: linuxBridgeNetworkName2,
 			}
-			previousVMTemplateSpec := vm.Spec.Template.Spec.DeepCopy()
 
 			Expect(kubevirt.Client().VirtualMachine(vm.Namespace).RemoveInterface(context.Background(), vm.Name, removeIfaceOpts)).To(Succeed())
 
-			By("wait for requested interface VM spec have 'absent' state")
-			Eventually(func() v1.InterfaceState {
+			By("wait for requested interface be cleared from VM spec")
+			Eventually(func(g Gomega) {
 				var err error
 				vm, err = kubevirt.Client().VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, &metav1.GetOptions{})
 				Expect(err).NotTo(HaveOccurred())
+
 				iface := vmispec.LookupInterfaceByName(vm.Spec.Template.Spec.Domain.Devices.Interfaces, linuxBridgeNetworkName2)
-				return iface.State
-			}, 30*time.Second).Should(Equal(v1.InterfaceStateAbsent))
+				g.Expect(iface).To(BeNil(), "absent iface should be removed from stopped VM spec")
+			}, 30*time.Second).Should(Succeed())
 
-			Expect(previousVMTemplateSpec.Networks).To(Equal(vm.Spec.Template.Spec.Networks), "network spec should not change")
+			iface := libnet.LookupNetworkByName(vm.Spec.Template.Spec.Networks, linuxBridgeNetworkName2)
```

Comment:
I do not think we remove the network, otherwise the flake would have been constantly failing, not like now.

Reply:
Networks are expected to be removed as well https://github.com/kubevirt/kubevirt/blob/release-1.0/pkg/virt-controller/watch/vm.go#L2663-L2665.
I think now it doesn't even reach the spec networks assert.

Reply:
I think I now understand why it worked before and it is flaky.
The flake was due to the fact that we catched the interface before it was removed and therefore the network was still there.

So these two come as pairs.